Doc/tools/touch-clean-files.py

```diff
@@ -6,16 +6,27 @@
 
 from pathlib import Path
 
-# Input file has blank line between entries to reduce merge conflicts
-with Path("Doc/tools/clean-files.txt").open() as clean_files:
-    CLEAN = [
+# Exclude these whether they're dirty or clean,
+# because they trigger a rebuild of dirty files.
+EXCLUDES = {
+    Path("Doc/includes/wasm-notavail.rst"),
+    Path("Doc/whatsnew/changelog.rst"),
+}
+
+# Exclude "Doc/venv/"
+ALL_RST = {rst for rst in Path("Doc/").rglob("*.rst") if rst.parts[1] != "venv"}
+
+with Path("Doc/tools/dirty-files.txt").open() as clean_files:
+    DIRTY = {
         Path(filename.strip())
         for filename in clean_files
         if filename.strip() and not filename.startswith("#")
```

Comment:
Could possibly reduce repetition here with a walrus, but feel free to ignore if you don't like it:

```suggestion
        Path(filename)
        for line in clean_files
        if (filename := line.strip()) and not filename.startswith("#")
```

Reply:
IMO, I personally find the latter a lot less easy to read and reason about in this context, as the assignment to `filename` is (physically) located well after its first use, in a not so obvious place that readers (and tools) might not immediately think to look, and the dependency of the item value rather then presence on the `if` block adds significant cognitive complexity, at least for me.